### ISSUE TITLE
[en] Support for non-standard pronouns (aka Neopronouns)

### ISSUE THREAD

**Issue by solonovamax** [languagetool-org/languagetool#4837]
TL;DR of this issue is I think LanguageTool should add support for non-standard pronouns, or neopronouns.

-------

For anyone who doesn't know what "neopronouns" are, they're alternative pronoun sets to the standard "he/him", "she/her", or "they/them". For example: "ko/kor/kors/kor’s/korself", "nya/nyan/nyans/nyans/nyanself", etc.
They are used the same as normal pronouns:
> This morning, **_nya_** went to the park.
> I went with **_nyan_**.
> And **_nya_** brought **_nyans_** frisbee.
> At least I think it was **_nyans_**.
> By the end of the day, **_nya_** started throwing the frisbee to **_nyanself_**.

I won't turn this into a whole politics thing, but if you want to read more, [check](https://nonbinary.miraheze.org/wiki/Pronouns) [out](https://www.nytimes.com/2021/04/08/style/neopronouns-nonbinary-explainer.html) [these](https://lgbta.wikia.org/wiki/Neopronouns) [pages](https://www.quora.com/What-are-neopronouns-and-why-should-I-use-them).
But to summarize my opinion on them quickly: if they're not hurting anyone and make people happy, then what's the harm in accepting them?

But, as society has become more inclusive over the past few years, the usage of neopronouns has increased significantly.
Beacuse of this, I think it'd be appropriate if LanguageTool had support for people who use neopronouns.
It would also be great if there was correction for misuse of neopronouns, as sometimes those who are unfamiliar with them might accidentally use them incorectly.
Though, as not everyone is ready for them yet, I think it'd be best as an opt-in rule, rather than opt-out.

Additionally, because there are such a wide variety of neopronouns, I think it'd be best to support a few of the more common ones, and then allow users to add their own, using an API similar to the `/words/*` API.

I personally believe that this would be a good feature because:
- It has no downsides (Other than the added complexity. But I'm willing to contribute code.)
- It's inclusive towards people a wider array of people
- I don't think it would require *that* much changing to the rules. Other than that you might need to introduce some new construct that allows words to be injected into the rules at runtime.

<details>
<summary>Examples of Neopronouns</summary>

If you want some examples of neopronouns, here are a few sites have lists (I also provided a command that can be run in bash to get a semi-usable list of the data)

[Pronouny](https://pronouny.xyz/pronouns/list/public): `curl 'https://pronouny.xyz/api/pronouns' 2>/dev/null | jq -Sc '.[] | select(.pattern != null) | {pattern:.pattern, details:{subject:.subject, object:.object, determiner:.determiner, possessive:.possessive, reflexive:.reflexive, pronounType:.pronounType}}' -C | sort | uniq`

[Neopronoun list Carrd](https://neopronounlist.carrd.co/): `curl 'https://neopronounlist.carrd.co/#'"$i" 2>/dev/null | htmlq -pt '#list01 > ul > li > p' | sed -Ee 's/\s*\((only to be|not to be).*\)//g' -e "s/’/'/g" | sort | uniq -i`

[LGBTA Wiki](https://lgbta.wikia.org/wiki/Neopronouns): `curl 'https://lgbta.wikia.org/wiki/Neopronouns' 2>/dev/null | htmlq -tp '#mw-content-text > div > table > tbody > tr > td:nth-child(2)' | perl -077 -pe 's/(?m)(^.*$)\n\n(^.*$)\n\n(^.*$)\n\n(^.*$)\n\n(^.*$)\n\n/$1\/$2\/$3\/$4\/$5:\n -> nominative: $1, accusative: $2, pronominal possessive: $3, predicative possessive: $4, reflexive: $5\n/g'`

(Note: some of these sites may provide pronouns that I do not personally agree with. Such as pronouns relating to real people, like the YouTuber Dream. Along with that, there are emoji prounouns, which I have not yet decided where I stand for them. They are just given to provide some examples of neopronouns that people might use.)
</details>

Now, I also understand that this might be a more "controversial" feature to add.
The main reason being, that there are many people who don't support the use of them.
But, people can just leave the rule disabled if they don't wish to use them. Nothing would force them into using them.
